tests/functional/test_globals.py

```diff
@@ -0,0 +1,21 @@
+import os
```

Comment:
In addition to a functional test, it may be worth adding a unit test for the `GlobalOptionsDocumenter` class that would:

* Use the test's own controlled arg table (e.g. not sourced from the `cli.json` file). 
* Assert the content for both the global options and synopsis are correct for arguments that:
   * Are of type string
   * Are of type int
   * Are of type boolean
   * Have enums

The main reason that I'm suggesting adding these tests is that if we ever decide to refactor this logic, we have some basic tests in place that we can update or port refactor to make sure our abstraction is working correctly. For example, if we decided to do away with storing the content to a file and we remove those functional tests, we would not have any (direct or indirect) tests capturing the behavior of what are abstraction should be behaving.